Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1745,11 +1731,11 @@ private InvalidBlockException ValidateNextBlock(Block<T> nextBlock)
                     message
                 );
             }
-            else if (Tip is { } tip && actualProtocolVersion < tip.ProtocolVersion)
+            else if (Count > 0 && actualProtocolVersion < Tip.ProtocolVersion)
```

Comment:
Isn't the added condition (`Count > 0`) always `true`?

Reply:
Actually no. When creating `BlockChain<T>`, given genesis block is "appended" instead of being set via `Append()` where `ValidateNextBlock()` is called, in which case the blockchain can be empty. Removing this results in an error since there is no `Tip` to retrieve.

Reply:
Fair enough.  I suggest to adding a private property like…

```csharp
private BeingInstantiated => Count < 1;
```

and replacing such predicates with `BeingInstantiated` so that other contributors easily can be aware of their intention.

Reply:
Eh, still feels a bit icky.

I agree using something like `BeingInstantiated` as a private property makes the code clearer locally. Half way through implementing your suggestion, however, I realized the condition of being instantiated is not the same as `Count` being zero. Implementing with a private field `_beingInstantiated` and actually setting the flag before exiting a constructor can do the job instead, but this has its own set of problems of making the code more prone to mistakes. I can't quite think of a better yet simple solution. I think method overloading or some kind of redesign of `Append()` might be preferable, in which case, the scope of this PR gets too big. I'll try to get back on this if I get the chance.

Reply:
Okay, then please leave comments on such predicates to provide some context.